### PR TITLE
feat: Add support for v2 api

### PR DESCRIPTION
**Pull request by paul-paliychuk**

### CODE REVIEW

**Review by petergarbers — APPROVED** [review]

**Comment by paul-paliychuk** [comment]
Pending fixing tests + reorganizing langchain imports

File: src/langchain/message_history.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+}
+
+/**
+ * Class used to manage the memory of a chat session, including loading
+ * and saving the chat history, and clearing the memory when needed. It
+ * uses the ZepClient to interact with the Zep service for managing the
+ * chat session's memory.
+ *
+ * @example
+ * ```typescript
```
Comment (petergarbers) [inline src/langchain/message_history.ts:34]: Remove these because we have examples and documentation

File: src/langchain/vector_store.ts [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+
+   /**
+    * Creates a new document collection.
+    *
+    * @param {IZepConfig} args - The configuration object for the Zep API.
+    */
+   private async createCollection(args: IZepConfig) {
+      if (!args.embeddingDimensions) {
+         throw new Error(`Collection ${args.collectionName} not found. 
+ You can create a new Collection by providing embeddingDimensions.`);
```
Comment (petergarbers) [inline src/langchain/vector_store.ts:145]: DOn't forget this

File: src/memory_manager.ts [hunk 3]
```diff
@@ -43,6 +44,29 @@ export default class MemoryManager {
       return new Session(responseData);
    }
 
+   async synthesizeQuestion(
+      sessionId: string,
+      lastN: number = 3,
+   ): Promise<string> {
+      if (sessionId === null || sessionId.trim() === "") {
```
Comment (petergarbers) [inline src/memory_manager.ts:51]: Check for the empty string instead of trim

**Pull request merged**